src/test/java/uk/gov/ons/ctp/response/action/service/ActionServiceTest.java

```diff
@@ -194,80 +212,76 @@ public void testUpdateActionNoUpdate() {
   }
 
   @Test
-  public void testCreateScheduledActionsWithoutAnActionPlanJobNothingHappens() {
-    actionService.createScheduledActions(1);
-
-    assertEquals(0, actionRepo.count());
-  }
-
-  @Test
-  public void testCreateScheduledActionsWithNoActiveCaseDoesNotCreateActions() {
-    //// Given
-    // Create Action Plan
-    final int actionPlanPK = 1;
-    final ActionPlan actionPlan = mock(ActionPlan.class);
-
-    // Create Action Plan Job for Action Plan
-    final int actionPlanJobPK = 1;
-    final ActionPlanJob actionPlanJob = mock(ActionPlanJob.class);
-    when(actionPlanJob.getActionPlanFK()).thenReturn(actionPlanPK);
-
-    when(actionPlanJobRepository.findByActionPlanJobPK(actionPlanJobPK)).thenReturn(actionPlanJob);
-    when(actionPlanRepository.findByActionPlanPK(actionPlanPK)).thenReturn(actionPlan);
-
-    when(actionRepo.findPotentialActionsActiveDate(eq(actionPlanPK), any(Timestamp.class)))
-        .thenReturn(new ArrayList<>());
+  public void testCreateScheduledActions() {
 
-    //// When
-    actionService.createScheduledActions(actionPlanJobPK);
+    // Given
+    when(actionCaseRepo.findByActionPlanFK(any()))
+        .thenReturn(Collections.singletonList(actionCases.get(0)));
+    when(actionRuleRepo.findByActionPlanFK(any()))
+        .thenReturn(Collections.singletonList(actionRules.get(0)));
+    when(actionTypeRepo.findByActionTypePK(any())).thenReturn(actionTypes.get(0));
 
-    //// Then
-    // action plan has been updated
-    verify(actionPlan, times(1)).setLastRunDateTime(any(Timestamp.class));
+    // When
+    ActionPlan actionPlan = new ActionPlan();
+    actionPlan.setActionPlanPK(1);
+    actionService.createScheduledActions(actionPlan, actionPlanJob);
 
-    // action plan job has been updated
-    verify(actionPlanJob, times(1)).complete(any(Timestamp.class));
+    // Then
+    verify(actionRepo, times(1)).saveAndFlush(any());
+    verify(actionPlanRepo, times(1)).saveAndFlush(any());
+    verify(actionPlanJobRepo, times(1)).saveAndFlush(any());
+  }
 
-    // did not attempt to create actions
-    verify(actionRepo, times(0)).save(any(Action.class));
-    verify(actionRepo, times(1)).flush();
+  @Test
+  public void testCreateScheduledActionsBCaseRule5() {
```

Comment:
Does "Rule 5" mean PK = 5? We probably shouldn't hard-code primary keys into our test names etc.